File: collect_app/src/main/java/org/odk/collect/android/configure/qr/QRCodeTabsAdapter.java
package org.odk.collect.android.configure.qr;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class QRCodeTabsAdapter extends FragmentStateAdapter {
    public QRCodeTabsAdapter(FragmentActivity fa) {
        super(fa);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position) {
            case 0:
                return new QRCodeScannerFragment();
            case 1:
                return new ShowQRCodeFragment();
            default:
                // should never reach here
                throw new IllegalArgumentException("Fragment position out of bounds");
        }
    }

    @Override
    public int getItemCount() {
        return 2;
    }
}

File: collect_app/src/main/java/org/odk/collect/android/configure/qr/QRCodeTabsAdapter.kt
package org.odk.collect.android.configure.qr

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import java.lang.IllegalArgumentException

class QRCodeTabsAdapter(activity: FragmentActivity) : FragmentStateAdapter(activity) {

    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> QRCodeScannerFragment()
            1 -> ShowQRCodeFragment()
            else -> throw IllegalArgumentException("Fragment position out of bounds")
        }
    }

    override fun getItemCount(): Int {
        return 2
    }
}
